File: src/main/java/com/teste/dev/java/infuse/service/PedidoService.java
package com.teste.dev.java.infuse.service;

import com.teste.dev.java.infuse.dto.PedidoDTO;
import com.teste.dev.java.infuse.dto.PedidoInputDTO;
import com.teste.dev.java.infuse.model.Pedido;
import org.springframework.http.ResponseEntity;

import java.text.ParseException;
import java.util.List;
import java.util.Optional;

public interface PedidoService {

    ResponseEntity<PedidoInputDTO> cadastrarPedido(PedidoInputDTO pedidos);

    Optional<Pedido> buscarPedidoPeloNumero(Integer numeroControle);

    List<PedidoDTO> buscarPedidos(Integer numero, String dataCadastro, Boolean todos) throws ParseException;
}

File: src/main/resources/static/app/service/PedidoService.js
;(function() {
    angular
        .module('app')
        .service('PedidoService', ['$http', function($http) {
            return {
                get: function(todos, dataCadastro, numero) {
                    return $http.get(`/pedidos/buscar?todos=${todos}&dataCadastro=${dataCadastro}&numero=${numero}`);
                }
                ,
                save: function(data) {
                    return $http.post('/pedidos/novo', data);
                }
                ,
                getCliente: function() {
                    return $http.get('/clientes/buscar');
                }
            };
        }]);
})();
